esrally/mechanic/mechanic.py

```diff
@@ -616,26 +490,20 @@ def receiveUnrecognizedMessage(self, msg, sender):
             if isinstance(msg, ResetRelativeTime):
                 self.logger.info("Resetting relative time of system metrics store on host [%s].", self.host)
                 self.metrics_store.reset_relative_time()
-            elif isinstance(msg, OnBenchmarkStart):
-                self.mechanic.on_benchmark_start()
-                self.wakeupAfter(METRIC_FLUSH_INTERVAL_SECONDS)
-                self.send(sender, BenchmarkStarted())
             elif isinstance(msg, thespian.actors.WakeupMessage):
                 if self.running:
                     self.logger.debug("Flushing system metrics store on host [%s].", self.host)
                     self.metrics_store.flush(refresh=False)
                     self.wakeupAfter(METRIC_FLUSH_INTERVAL_SECONDS)
-            elif isinstance(msg, OnBenchmarkStop):
-                self.mechanic.on_benchmark_stop()
-                self.metrics_store.flush(refresh=False)
-                # clear metrics store data to not send duplicate system metrics data
-                self.send(sender, BenchmarkStopped(self.metrics_store.to_externalizable(clear=True)))
             elif isinstance(msg, StopNodes):
                 self.logger.info("Stopping nodes %s.", self.mechanic.nodes)
                 self.mechanic.stop_engine()
-                self.send(sender, NodesStopped(self.metrics_store.to_externalizable()))
+                self.metrics_store.flush(refresh=False)
+                self.send(sender, NodesStopped(self.metrics_store.to_externalizable(clear=True)))
                 # clear all state as the mechanic might get reused later
                 self.metrics_store.close()
+                # TODO: Run the reporter (StatsCalculator) here to calculate summary stats and safe them to the
```

Comment:
s/safe/save

Reply:
Addressed in https://github.com/elastic/rally/pull/779/commits/bdb66bfbea29a08a8557376f130151142d83b62b